external/kafka/src/main/scala/org/apache/spark/streaming/kafka/DirectKafkaInputDStream.scala

```diff
@@ -158,4 +166,37 @@ class DirectKafkaInputDStream[
     }
   }
 
+  private[streaming]
+  class DirectKafkaStreamingListener extends StreamingListener {
+
+    val offsetCommitEnabled = context.conf.getBoolean("spark.streaming.kafka.offsetCommit", false)
```

Comment:
So what is your suggestion? shall we set the mechanism as default and make it invisible to user?